src/chess/chess_game.rs

```diff
@@ -206,3 +314,211 @@ impl IGame for ChessGame {
+        "d8h8", "e8e1", "e8e2", "e8e3", "e8a4", "e8e4", "e8b5", "e8e5", "e8h5", "e8c6", "e8d6",
+        "e8e6", "e8f6", "e8g6", "e8c7", "e8d7", "e8e7", "e8f7", "e8g7", "e8a8", "e8b8", "e8c8",
+        "e8d8", "e8f8", "e8g8", "e8h8", "f8f1", "f8f2", "f8a3", "f8f3", "f8b4", "f8f4", "f8c5",
+        "f8f5", "f8d6", "f8e6", "f8f6", "f8g6", "f8h6", "f8d7", "f8e7", "f8f7", "f8g7", "f8h7",
+        "f8a8", "f8b8", "f8c8", "f8d8", "f8e8", "f8g8", "f8h8", "g8g1", "g8a2", "g8g2", "g8b3",
+        "g8g3", "g8c4", "g8g4", "g8d5", "g8g5", "g8e6", "g8f6", "g8g6", "g8h6", "g8e7", "g8f7",
+        "g8g7", "g8h7", "g8a8", "g8b8", "g8c8", "g8d8", "g8e8", "g8f8", "g8h8", "h8a1", "h8h1",
+        "h8b2", "h8h2", "h8c3", "h8h3", "h8d4", "h8h4", "h8e5", "h8h5", "h8f6", "h8g6", "h8h6",
+        "h8f7", "h8g7", "h8h7", "h8a8", "h8b8", "h8c8", "h8d8", "h8e8", "h8f8", "h8g8", "a7a8q",
+        "a7a8r", "a7a8b", "a7a8k", "a7b8q", "a7b8r", "a7b8b", "a7b8k", "b7a8q", "b7a8r", "b7a8b",
+        "b7a8k", "b7b8q", "b7b8r", "b7b8b", "b7b8k", "b7c8q", "b7c8r", "b7c8b", "b7c8k", "c7b8q",
+        "c7b8r", "c7b8b", "c7b8k", "c7c8q", "c7c8r", "c7c8b", "c7c8k", "c7d8q", "c7d8r", "c7d8b",
+        "c7d8k", "d7c8q", "d7c8r", "d7c8b", "d7c8k", "d7d8q", "d7d8r", "d7d8b", "d7d8k", "d7e8q",
+        "d7e8r", "d7e8b", "d7e8k", "e7d8q", "e7d8r", "e7d8b", "e7d8k", "e7e8q", "e7e8r", "e7e8b",
+        "e7e8k", "e7f8q", "e7f8r", "e7f8b", "e7f8k", "f7e8q", "f7e8r", "f7e8b", "f7e8k", "f7f8q",
+        "f7f8r", "f7f8b", "f7f8k", "f7g8q", "f7g8r", "f7g8b", "f7g8k", "g7f8q", "g7f8r", "g7f8b",
+        "g7f8k", "g7g8q", "g7g8r", "g7g8b", "g7g8k", "g7h8q", "g7h8r", "g7h8b", "g7h8k", "h7g8q",
+        "h7g8r", "h7g8b", "h7g8k", "h7h8q", "h7h8r", "h7h8b", "h7h8k",
+    ]
+    .into_iter()
+    .map(|s| {
+        assert!(s.len() == 4 || s.len() == 5);
+        let sf = chess::File::from_index(s.chars().nth(0).unwrap() as usize - 'a' as usize);
+        let sr = chess::Rank::from_index(s.chars().nth(1).unwrap() as usize - '1' as usize);
+        let df = chess::File::from_index(s.chars().nth(2).unwrap() as usize - 'a' as usize);
+        let dr = chess::Rank::from_index(s.chars().nth(3).unwrap() as usize - '1' as usize);
+        let source = chess::Square::make_square(sr, sf);
+        let dest = chess::Square::make_square(dr, df);
+        let promotion = if s.len() == 5 {
+            Some(match s.chars().nth(4).unwrap() {
+                'q' => chess::Piece::Queen,
+                'k' => chess::Piece::Knight,
+                'r' => chess::Piece::Rook,
+                'b' => chess::Piece::Bishop,
+                c => panic!("unknown char: {:?}", c),
+            })
+        } else {
+            None
+        };
+        return ChessMove::new(chess::ChessMove::new(source, dest, promotion));
+    })
+    .collect_vec()
+});
+
+static MOVE_TO_NN_INDEX: Lazy<Vec<usize>> = Lazy::new(|| {
+    let mut res = vec![0xffff; 64 * 64 + 22 * 4];

```

Comment:
did not you want something like `usize::MAX` here? instead of 0xffff

Reply:
64*64+22*4=4184= 0x1058, so we are very far from 0xffff, i forgot to change MOVE_TO_NN_INDEX to be Vec<u16>... currently it's 15Kb, which is not ideal